<launch>
  <arg name="namespace" default="project11/mesobot"/>
  <arg name="host_namespace"/>
  <arg name="usbl_address" default="2509"/>
  <arg name="usbl_backup_address" default="2712"/>
  <arg name="platform_name" default="mesobot"/>
  <arg name="with_ctd" default="false"/>

  <include file="$(find project11)/launch/robot_core.launch">
    <arg name="namespace" value="$(arg namespace)"/>
    <arg name="enableBridge" value="false"/>
  </include>

  <group ns="$(arg namespace)">
    <node pkg="mesobot_project11" type="mesobot_node.py" name="mesobot">
      <remap from="sms" to="/$(arg host_namespace)/usbl_modem/received_sms"/>
      <remap from="send_sms" to="/$(arg host_namespace)/usbl_modem/send_sms"/>
      <remap from="position" to="/$(arg host_namespace)/usbl_ranger/positions/$(arg usbl_address)"/>
      <remap from="backup_position" to="/$(arg host_namespace)/usbl_ranger/positions/$(arg usbl_backup_address)"/>
      <param name="beacon_id" value="$(arg usbl_address)"/>
    </node>

    <node if="$(arg with_ctd)" pkg="mesobot_project11" type="ctd.py" name="ctd">
      <remap from="raw" to="/$(arg host_namespace)/usbl_modem/raw"/>
      <remap from="send_sms" to="/$(arg host_namespace)/usbl_modem/send_sms"/>
    </node>

    <rosparam command="load" file="$(find mesobot_project11)/config/mesobot.yaml"/>

  </group>

  <include file="$(find project11)/launch/platform.launch">
    <arg name="namespace" value="$(arg namespace)"/>
    <arg name="platform_name" value="$(arg platform_name)"/>
    <arg name="platform_package" value="mesobot_project11"/>
  </include>

  <include file="$(find project11)/launch/platform_nav_source.launch">
    <arg name="namespace" value="$(arg namespace)"/>
    <arg name="platform_name" value="$(arg platform_name)"/>
    <arg name="position_topic" value="sensors/nav/pose"/>
    <arg name="orientation_topic" value="sensors/nav/pose"/>
  </include>

  <node pkg="mesobot_behavior" type="mesobot_behavior_node.py" name="mesobot_behavior" ns="$(arg host_namespace)"/>

</launch>
